spec/rubocop/cop/rspec/repeated_example_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::RSpec::RepeatedExample do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense for repeated example' do
+    expect_violation(<<-RUBY)
+      describe 'doing x' do
+        it "does x" do
+                    ^^ Don't repeat examples within an example group.
+          expect(foo).to be(bar)
+        end
+
+        it "does y" do
+                    ^^ Don't repeat examples within an example group.
+          expect(foo).to be(bar)
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register a violation if rspec keyword magic is involved' do
+    expect_no_violations(<<-RUBY)
+      describe 'doing x' do
+        it "does x" do
+          expect(foo).to be(bar)
+        end
+
+        it "does y", :magic do
```

Comment:
Nitpick: These are tags, not keywords. And I'd prefer using e.g. `:focus` instead of `:magic` to make it more "realistic".